go-controller/pkg/ovn/secondary_localnet_network_controller.go

```diff
@@ -353,11 +390,44 @@ func (oc *SecondaryLocalnetNetworkController) newRetryFramework(
 }
 
 func (oc *SecondaryLocalnetNetworkController) localnetPortNetworkNameOptions() map[string]string {
-	localnetLSPOptions := map[string]string{
-		"network_name": oc.GetNetworkName(),
+	return map[string]string{
+		"network_name": oc.getNetworkName(),
+	}
+}
+
+func (oc *SecondaryLocalnetNetworkController) checkBridgeMapping(obj interface{}) error {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return fmt.Errorf("could not cast %T object to *knet.Pod", obj)
+	}
+
+	if !util.PodScheduled(pod) {
+		return nil
+	}
+
+	openvSwitch, err := ovsops.GetOpenvSwitch(oc.ovsClient)
+	if err != nil {
+		return err
+	}
```

Comment:
Do we need to invoke this everytime we run `checkBridgeMapping` ? 
Couldn't we encapsulate this in some struct and just read the external IDs from it ? 

Just asking, I haven't checked the actual implementation of  `ovsops.GetOpenvSwitch` .

Reply:
from what i read we need it to get the new data, but i will make sure
the current places on code that use it, do call it as well fwiw

Reply:
interesting, with quick and dirty code, seems you are right